app/controllers/admin/routes.py

```diff
@@ -245,9 +245,15 @@ def eventDisplay(eventId):
         if event.recurringId:
             eventSeriesList = list(Event.select().where(Event.recurringId == event.recurringId).order_by(Event.startDate))
             eventIndex = eventSeriesList.index(event)
-            if len(eventSeriesList) != (eventIndex + 1):
-                eventData["nextRecurringEvent"] = eventSeriesList[eventIndex + 1]
-
+            # Find the next uncanceled event
+            nextUncanceledEvent = None
+            for nextEvent in eventSeriesList[eventIndex + 1:]:
+                if not nextEvent.isCanceled:  
+                    nextUncanceledEvent = nextEvent
+                    break
+            if nextUncanceledEvent:
+                eventData["nextRecurringEvent"] = nextUncanceledEvent
```

Comment:
Fixed

Reply:
I restored the previous code. We cannot exclude canceled events by using the where clause in the query because then canceled events would not be passed at all and that therefore forfeits the purpose of ability to view information about canceled event